sycl/include/CL/sycl/handler.hpp

```diff
@@ -660,16 +660,29 @@ class __SYCL_EXPORT handler {
         KernelFunc);
   }
 
-  /* 'wrapper'-based approach using 'NormalizedKernelType' struct is
-   * not applied for 'void(void)' type kernel and
-   * 'void(sycl::group<Dims>)'. This is because 'void(void)' type does
-   * not have argument to normalize and 'void(sycl::group<Dims>)' is
-   * not supported in ESIMD.
+  // For 'void' kernel argument (single_task)
+  template <class KernelType, typename ArgT, int Dims>
+  typename std::enable_if_t<std::is_same<ArgT, void>::value, KernelType *>
+  ResetHostKernel(const KernelType &KernelFunc) {
+    struct NormalizedKernelType {
+      KernelType MKernelFunc;
+      NormalizedKernelType(const KernelType &KernelFunc)
+          : MKernelFunc(KernelFunc) {}
+      void operator()(const nd_item<Dims> &Arg) {
+        detail::runKernelWithoutArg(MKernelFunc);
+      }
+    };
+    return ResetHostKernelHelper<KernelType, struct NormalizedKernelType, Dims>(
+        KernelFunc);
+  }
+
+  /* 'wrapper'-based approach using 'NormalizedKernelType' struct is not used
+   * for 'void(sycl::group<Dims>)' since 'void(sycl::group<Dims>)' is not
+   * supported in ESIMD.
```

Comment:
Is `sycl::group<Dims>` kernel supported at all (non-ESIMD case)? If not, then RT error would be better.

Reply:
My current understanding is the following: it's a "won't happen" case for ESIMD, hence no error.
Shall review this.

Reply:
Assuming something won't happen and just ignoring the possibility is not safe. Please add error reporting.

Reply:
That's true. This code was introduced previously and results in runtime failure in case of using esimd emu backend.
This is to be addressed in the following changes.